app/controllers/streams_controller.rb

```diff
@@ -25,6 +25,21 @@ def public
     stream_responder(Stream::Public)
   end
 
+  def local_public
+    return stream_responder(Stream::LocalPublic) if AppConfig.settings.enable_show_local_post_link == "always"
+
+    if AppConfig.settings.enable_show_local_post_link == "admin" && Role.is_admin?(current_user)
+      return stream_responder(Stream::LocalPublic)
+    end
+
+    if AppConfig.settings.enable_show_local_post_link == "moderator" &&
+       (Role.moderator?(current_user) || Role.is_admin?(current_user))
+      return stream_responder(Stream::LocalPublic)
+    end
+
+    stream_responder(Stream::Public)
+  end
```

Comment:
I think this should 404 instead of returning other content than promised.

With above change we could then write this as (untested):

```rb
if AppConfig.has_local_posts_stream?(current_user)
  stream_responder(Stream::LocalPublic)
else 
  render status: :404
end
```